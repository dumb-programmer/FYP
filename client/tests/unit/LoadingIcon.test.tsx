import React from "react";
import { render, screen } from "@testing-library/react";
import { expect, describe, it } from "vitest";
import LoadingIcon from "../../src/components/LoadingIcon";

const renderComponent = () => render(<LoadingIcon size={20} />);

describe("EditChatForm", () => {
    it("renders correctly", () => {
        renderComponent();

        const component = screen.getByTestId("loading-icon");

        expect(component).toMatchSnapshot();
    });
})