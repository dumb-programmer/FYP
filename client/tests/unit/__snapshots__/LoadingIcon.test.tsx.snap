// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditChatForm > renders correctly 1`] = `
<svg
  data-testid="loading-icon"
  height="20"
  preserveAspectRatio="xMidYMid"
  viewBox="0 0 100 100"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="35"
    stroke="#fff"
    stroke-dasharray="164.93361431346415 56.97787143782138"
    stroke-width="10"
  >
    <animatetransform
      attributeName="transform"
      dur="1s"
      keyTimes="0;1"
      repeatCount="indefinite"
      type="rotate"
      values="0 50 50;360 50 50"
    />
  </circle>
</svg>
`;
